onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -241,7 +235,11 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
     utils::DumpNodeInputs(op_kernel_context, p_op_kernel->Node());
 #endif
 
+    std::string node_name;
```

Comment:
`node_name` is a little generic. i.e. it's not obvious that it's different from `node.Name()`.

how about something like `node_name_for_profiling`?

and we can do something like:

```
const std::string node_name_for_profiling = [&]() -> std::string {
  if (!is_profiler_enabled) return {};
  return ...
}();
```